app.rb

```diff
@@ -0,0 +1,135 @@
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'book'
+require_relative 'rental'
+require_relative 'classroom'
+
+class App
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def start_console
+    puts 'welcome to school Library App!'
+    until list_of_options
+      input = gets.chomp
+      if input == '7'
+        puts 'Thank You for using our school Library!'
+        break
+      end
+
+      option input
+    end
+  end
+
+  def list_all_books
+    puts 'Please add a book to the library' if @books.empty?
+    @books.each { |book| puts "Title: #{book.title}, Author: #{book.author}" }
+  end
+
+  def list_all_persons
+    puts 'Please add a person to the library' if @people.empty?
+    @people.each { |person| puts "[#{person.class.name}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}" }
+  end
+
+  def create_person
+    puts 'Do you want to create a student (1) or a teacher (2)? [Input the number]:'
+    option = gets.chomp
+    case option
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'Invalid option'
+      nil
+    end
+  end
+
+  def create_student
+    puts 'create a new student'
+    puts 'Age:'
+    age = gets.chomp.to_i
+    puts 'Name:'
+    name = gets.chomp
+    puts 'Has parent permission? [Y/N]:'
+    parent_permission = gets.chomp.downcase
+
+    case parent_permission
+    when 'y'
+      student = Student.new(classroom: @classroom, age: age, name: name, parent_permission: true)
+      @people << student
+      puts 'Student created successfully'
+    when 'n'
+      student = Student.new(classroom: @classroom, age: age, name: name, parent_permission: false)
+      @people << student
+      puts 'Student created successfully'
+    else
+      puts 'Invalid option'
+      nil
+    end
+  end
+
+  def create_teacher
+    puts 'create a new teacher'
+    puts 'Age:'
+    age = gets.chomp.to_i
+    puts 'Name:'
+    name = gets.chomp
+    puts 'Specialization:'
+    specialization = gets.chomp
+    teacher = Teacher.new(age, name, specialization)
```

Comment:
- [ ] when creating a new teacher you are sending the age, name and specialization as arguments in that order, however, if you go to the teacher class you can see that it takes the argument in a different order therefor the application, at the moment, is saving the specialization as the name of the teacher. to solve this you can just switch the arguments you sent like this:
```
teacher = Teacher.new(age, specialization, name)
```